File: JAVA-Basic/JavaBasic/src/WhileLoop/Exercise/Vacation.java
package WhileLoop.Exercise;

import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        double holidayPrice = Double.parseDouble(scan.nextLine());
        double budget = Double.parseDouble(scan.nextLine());
        int dayCounter = 0;
        int spendDayInRaw = 0;

        while (spendDayInRaw < 5 && budget < holidayPrice) {
            String currentAct = scan.nextLine();
            double currentSum = Double.parseDouble(scan.nextLine());
            dayCounter++;
            if (currentAct.equals("spend")) {
                budget -= currentSum;
                if (budget < 0) {
                    budget = 0;
                }
                spendDayInRaw++;
            } else if (currentAct.equals("save")) {
                budget += currentSum;
                spendDayInRaw = 0;
            }
        }
        if (spendDayInRaw == 5) {
            System.out.println("You can't save the money.");
            System.out.println(dayCounter);
        } else {
            System.out.printf("You saved the money for %d days.", dayCounter);
        }
    }
}

File: 01.JS-Basic/WHILE-ЦИКЪЛ Упражнение/Vacation.js
function vacation(input) {

    let holidayPrice = Number(input[0]);
    let ownMoney = Number(input[1]);
    let currentMoney = ownMoney;
    let index = 2;
    let spend = 0;
    let days = 0;

    while (true) {

        let operation = String(input[index])

        if (spend == 5) {
            console.log("You can't save the money.")
            console.log(days)
            break;
        }

        if (index % 2 == 0) {
            days++
            switch (operation) {
                case "spend": currentMoney -= Number(input[index + 1])
                    spend++
                    if (currentMoney < 0) {
                        currentMoney = 0
                    };
                    break;
                case "save": currentMoney += Number(input[index + 1])
                    spend = 0;
                    break;
            }
        }
        if (currentMoney >= holidayPrice) {
            console.log(`You saved the money for ${days} days.`)
            break;
        }
        index++
    }
}
vacation(["250",
    "150",
    "spend",
    "50",
    "spend",
    "50",
    "save",
    "100",
    "save",
    "100"])
